mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -393,11 +383,18 @@ private[ml] trait GBTParams extends TreeEnsembleParams with HasMaxIter {
   /** @group setParam */
   def setMaxIter(value: Int): this.type = set(maxIter, value)
 
-  /** @group setParam */
-  def setStepSize(value: Double): this.type = set(stepSize, value)
-
-  /** @group getParam */
-  final def getStepSize: Double = $(stepSize)
+  /**
+   * Step size (a.k.a. learning rate) in interval (0, 1] for shrinking the contribution of each
+   * estimator.
+   * (default = 0.1)
+   * @group setParam
+   */
+  def setStepSize(value: Double): this.type = {
+    if (!ParamValidators.inRange(0, 1, lowerInclusive = false, upperInclusive = true)(value)) {
+      throw new IllegalArgumentException(s"GBT parameter stepSize given invalid value $value.")
+    }
+    set(stepSize, value)
```

Comment:
@yanboliang Putting the check in `setStepSize` is not sufficient because users can bypass it by specifying the value in other places, e.g., `gbt.fit(df, gbt.stepSize -> 0.5)`. We need to put the check in `validateParams`.